### PR TITLE
social and email login/signup, restructured project, global saga

### PR DESCRIPTION
**Pull request by JohnAllen**

### CODE REVIEW

**Comment by JohnAllen** [comment]
How do you want to review this @KingJS226 ? I know you spent lots of time on it.  Looks like I am going to want to change some things. I don't want to make you work more than you have though.

**Comment by KingJS226** [comment]
you can add me to your slack with my email guidoxhindoli1@gmail.com.

File: app/containers/User/Login.js [hunk 1]
```diff
@@ -40,15 +73,81 @@ const Login = props => {
...
+                      redirect="/seo"
+                      onLoginSuccess={handleSocialLogin}
+                      onLoginFailure={handleSocialLoginError}
+                    >
+                      Login with Amazon
+                    </SocialLoginButton>
+                  </div>
+                  <div className="form-group col-md-12 addAccountButton">
+                    <Link to="/signup">
+                      <span>create an account</span>
```
Comment (JohnAllen) [inline app/containers/User/Login.js]: Change to "SignUp" for consistency

File: app/containers/User/SignUp.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                    type="text"
+                    className="form-control"
+                    placeholder="Your Email *"
+                    name="email"
+                    value={values.email}
+                    onChange={handleInputChange}
+                    required
+                  />
+                </div>
+                <div className="form-group">
```
Comment (JohnAllen) [inline app/containers/User/SignUp.js]: remove this domain stuff from here

File: app/containers/User/SignUp.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+                      required
+                    />
+                  </div>
+                </div>
+                <h4 className="passwordError">{passwordError}</h4>
+                <h6 className="passwordError">{props.errorMessage}</h6>
+                <button type="submit" className="btnSubmit align-self-center">
+                  Sign Up
+                </button>
+              </div>
```
Comment (JohnAllen) [inline app/containers/User/SignUp.js]: Really signup should include social login as well; we allow users to use social for signup and login.

File: app/containers/User/Login.js [hunk 4]
```diff
@@ -40,15 +73,81 @@ const Login = props => {
...
+                      onLoginSuccess={handleSocialLogin}
+                      onLoginFailure={handleSocialLoginError}
+                    >
+                      Login with Facebook
+                    </SocialLoginButton>
+                  </div>
+                  <div className="form-group col-md-4">
+                    <SocialLoginButton
+                      provider="amazon"
+                      appId="amzn1.application-oa2-client.9ccf20a1f3cf4730a7a83e56203869c7"
```
Comment (JohnAllen) [inline app/containers/User/Login.js]: I also get this: 
Screen Shot 2019-10-21 at 10 29 19 AM

Is this another HTTPS thing do you think?

File: app/containers/User/SignUp.js [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+                      className="form-control"
+                      placeholder="Confirm Password *"
+                      name="confirmPassword"
+                      value={values.confirmPassword}
+                      onChange={handleInputChange}
+                      required
+                    />
+                  </div>
+                </div>
+                <h4 className="passwordError">{passwordError}</h4>
```
Comment (JohnAllen) [inline app/containers/User/SignUp.js]: make sure user can return/ click back to return to login page

File: app/containers/User/SignUp.js (hunk 2 above)
Comment (JohnAllen) [reply app/containers/User/SignUp.js]: like remove the entire form

File: app/containers/User/Login.js (hunk 4 above)
Comment (KingJS226) [reply app/containers/User/Login.js]: Yes, amazon sdk is working in only https. So this error appeared I believe.

**Pull request merged**